📝 Update prompts and German translation

nilslindemann opened the pull request:
_Setting it to draft temporarily to run one final language translation on the current state, to test if the prompts are good._

### What I have done 💪 

I have let the AI retranslate all the German documents, at least twice per document, and at least once with [GPT 5](https://platform.openai.com/docs/models/gpt-5). In the first run I mostly used [GPT 4o](https://platform.openai.com/docs/models/chatgpt-4o-latest) and a few times GPT 5, with prompt modifications in-between, learning from GPTs output (and vice versa I guess). In the second run, which was a batch run without prompt modifications in-between, I used GPT 5 (Plot twist, it does the job better <sup>[1]</sup>). I diffed both against the English document and the previous translation, sometimes doing global changes in the German docs on the way. I did not look at parts which were white in the _previous-translation_ ↔ _new translation_ diff (= unchanged parts), as I translated most of them a few months ago, reviewing them back then. But I used white parts in the _English document_ ↔ _new translation_ diff, to see if critical changes in the _previous-translation_ ↔ _new translation_ diff, like changes in code includes, are good and bring things in sync.

### What the PR changes 🔨 

This PR applies all changes of #13993 (merged), #13968, #13931, #13929 (closed, will be done here), #13930 (closed, will be done here) before it adds changes on top. I later apply @svlandeg's suggestions in the permalink PR (#13993) and manually fix outdated permalinks. The second batch run showed that the permalinks are in sync. The PR makes:

* heavy changes to the general prompt in the `translate.py`  
  * **Edit:** Yurii applied them in #13968
* heavy changes to `docs/de/llm-prompt.md`, which is the German LLM prompt, which gets appended to the general prompt in the `translate.py`.
* lots of changes to the German documents under `docs/de/docs`, to bring them in sync with the English docs (adding a few `abbr`s on top)

#### Outdated

* <strike>It also adds/fixes some things in the code of `translate.py`</strike>
  * **Edit:** Let's handle these in another PR. they are not relevant for this PR.
* <strike>a few changes to English documents under `docs/en/docs`, mostly fixes</strike>.
  * **Edit:** I outsourced this to #14048. Those changes are assumed in the translations here, so let's resolve that other PR first.
* <strike> I have added a file `docs/en/docs/_llm-test.md`, which can be used to test the prompt. The German translation of it is also included</strike>
  * **Edit:** Let's do this in  #14049.

### How to review 💪  

The Commits start clean, but sometimes [get a bit messy towards the end](https://xkcd.com/1296/), especially those made late at night 😅. Sorry about that. I suggest to not go through the PR commit-by-commit, but all-in-one, by using the _Files changed_ tab, because some files are changed more than once in more than one commit, sometimes overwriting changes of previous commits. I suggest directly comparing those German translations which have many changes with the English document.

### The result 😲 

After this PR, all <strike>111</strike> 112 (I have added a test document) translatable English documents have their in-sync German translation. 

### Credits 🤝 

Thanks to @tiangolo, @YuriiMotov, @alejsdev and everyone else for their friendly support.

### Footnote 📓 

<sup>[1]</sup> GPT 5's changes are approximately 90% fixes and good changes, approximately 5% are unnecessary changes and approximately 5% are changes which have to be manually edited or reverted. GPT 4 does more unnecessary changes, resulting in, on average, twice as big diffs. GPT 5 understands complex prompt instructions better than GPT 4 and applies them more precise.

nilslindemann committed:
Apply branch `update-translations-prompt`

nilslindemann committed:
Apply branch `de-prompt`

nilslindemann committed:
Apply branch `translations-de`

nilslindemann committed:
Apply branch `update-translations-de-ai`

nilslindemann committed:
Apply branch `add-permalinks-to-en-docs`

nilslindemann committed:
Add an example to the translate.py

So that the LLM really does not do that (it sometimes did)

nilslindemann committed:
A few fixes in the English docs

nilslindemann committed:
A few fixes in the German docs

nilslindemann committed:
Add/repair hashes in a file

More to come soon

nilslindemann committed:
Update alternatives.md

nilslindemann committed:
Update async.md + prompt

nilslindemann committed:
Update benchmarks.md

nilslindemann committed:
Update environment-variables.md

nilslindemann committed:
Update fastapi-cli.md

nilslindemann committed:
Intermezzo: `Typ-Annotation` --> `Typannotation`

nilslindemann committed:
Update features.md

nilslindemann committed:
Update help-fastapi.md

nilslindemann committed:
Update history-design-future.md

nilslindemann committed:
Update index.md, project-generation.md

index.md is already up to date, no changes

nilslindemann committed:
Update python-types.md

nilslindemann committed:
Update virtual-environments.md

nilslindemann committed:
Update about/index.md

nilslindemann committed:
Update additional-responses.md

nilslindemann committed:
Update advanced/additional-status-codes.md

The previous commit message should have been:

Update advanced/additional-responses.md

Time to make a break XD

nilslindemann committed:
Update advanced-dependencies.md

nilslindemann committed:
Update advanced/async-tests.md

The previous commit title should again have been:

Update advanced/advanced-dependencies.md

nilslindemann committed:
Update advanced/behind-a-proxy.md

nilslindemann committed:
Update advanced/custom-response.md

nilslindemann committed:
Update advanced/dataclasses.md

nilslindemann committed:
Update advanced/events.md

nilslindemann committed:
Update advanced/generate-clients.md + prompt

nilslindemann committed:
Update advanced/index.md

nilslindemann committed:
Update advanced/middleware.md

nilslindemann committed:
Update advanced/openapi-callbacks.md

nilslindemann committed:
Update advanced/openapi-webhooks.md

nilslindemann committed:
Update advanced/path-operation-advanced-configuration.md

nilslindemann committed:
Update advanced/response-change-status-code.md

nilslindemann committed:
Update advanced/response-cookies.md

nilslindemann committed:
Update advanced/response-directly.md

nilslindemann committed:
Update advanced/response-headers.md

nilslindemann committed:
Update advanced/settings.md

nilslindemann committed:
update advanced/sub-applications.md

nilslindemann committed:
Update advanced/templates.md

nilslindemann committed:
Update advanced/testing-dependencies.md + prompt

nilslindemann committed:
Update advanced/testing-events.md + prompt

nilslindemann committed:
Update advanced/testing-websockets.md

nilslindemann committed:
Update advanced/using-request-directly.md

nilslindemann committed:
Update advanced/websockets.md

nilslindemann committed:
Update advanced/wsgi.md

nilslindemann committed:
Apply recent changes in branch `add-permalinks-to-en-docs`

Update two German files accordingly (rest will be done when translating the specific files)

nilslindemann committed:
Update advanced/security/http-basic-auth.md

nilslindemann committed:
Remove typo in the German prompt

nilslindemann committed:
Formulate an instruction more precise in the German prompt

nilslindemann committed:
Update advanced/security/index.md

nilslindemann committed:
Update advanced/security/oauth2-scopes.md

nilslindemann committed:
Intermezzo: `check` --> `testen`

Smoother than `überprüfen`.

nilslindemann committed:
Update deployment/cloud.md

nilslindemann committed:
Update deployment/concepts.md + prompt

nilslindemann committed:
Update deployment/docker.md + prompt

nilslindemann committed:
Update deployment/https.md

nilslindemann committed:
Update deployment/index.md

nilslindemann committed:
Update deployment/manually.md

nilslindemann committed:
Update deployment/server-workers.md

nilslindemann committed:
Update deployment/versions.md

nilslindemann committed:
Update how-to/conditional-openapi.md

nilslindemann committed:
Update how-to/configure-swagger-ui.md

nilslindemann committed:
Update how-to/custom-docs-ui-assets.md + prompt

This adds a new rule to the prompt, to translate text inside title attributes of `abbr` elements. (which it did not do in a case in this document)

nilslindemann committed:
Update how-to/custom-request-and-route.md

nilslindemann committed:
Update how-to/extending-openapi.md

nilslindemann committed:
Update how-to/general.md

nilslindemann committed:
Update how-to/graphql.md

nilslindemann committed:
Update how-to/index.md

nilslindemann committed:
Update how-to/separate-openapi-schemas.md

nilslindemann committed:
Update how-to/testing-database.md + general prompt

In this document it added language codes to urls starting with "https://sqlmodel.tiangolo.com/", resulting in broken links.

So I changed the instructions in the translate.py, so that it only adds the language code when the URL starts exactly with "https://fastapi.tiangolo.com".

nilslindemann committed:
Update learn/index.md

nilslindemann committed:
Update resources/index.md

nilslindemann committed:
Update tutorial/background-tasks.md

nilslindemann committed:
Update tutorial/bigger-applications.md

nilslindemann committed:
Update tutorial/body-fields.md

nilslindemann committed:
Update tutorial/body-multiple-params.md

nilslindemann committed:
Update tutorial/body-nested-models.md

nilslindemann committed:
Update tutorial/body-updates.md

nilslindemann committed:
Intermezzo: Add a rule to the general prompt

..., which tells the LLM to fix wrongly translated anchor parts in links.

Back in those days I translated these parts, to link to the correct German heading, which then had a different anchor than the English heading. But now, as we use the same anchors
documentation-wide, this needs to be undone. The LLM missed some of these translated anchors, so I added this rule.

I will check via script, when gone through the whole documentation, if all those anchor parts are in sync and valid.

nilslindemann committed:
Intermezzo: make two rules in the German prompt mor precise

Before it seemingly randomly applied them, indicating that it has not understood them the way they were formulated.

Change two related things in two translations.

Remove the part to not translate link targets, as this may interfere with the more precise rulings in the general prompt.

nilslindemann committed:
Update tutorial/body.md + prompt

nilslindemann committed:
Update tutorial/cookie-param-models.md

nilslindemann committed:
Update tutorial/cookie-params.md

nilslindemann committed:
Update tutorial/cors.md

nilslindemann committed:
Update tutorial/debugging.md

nilslindemann committed:
Update tutorial/encoder.md

nilslindemann committed:
Update tutorial/extra-data-types.md

nilslindemann committed:
Update tutorial/extra-models.md

nilslindemann committed:
Update tutorial/first-steps.md

nilslindemann committed:
Update tutorial/handling-errors.md

nilslindemann committed:
Update tutorial/header-param-models.md

nilslindemann committed:
update tutorial/header-params.md + prompt

nilslindemann committed:
Update tutorial/index.md

nilslindemann committed:
Update tutorial/metadata.md + prompt

nilslindemann committed:
Update tutorial/middleware.md

nilslindemann committed:
Intermezzo: Format examples like in the general prompt

Removes one level of Use–mention distinction [1] by writing the examples in separate paragraphs, instead of wrapping them in backticks. This because the LLM still seems to ignore the rules sometimes, here it ignored the rule to translate the content of title attribute of abbr elements.

[1]: https://en.wikipedia.org/wiki/Use%E2%80%93mention_distinction

nilslindemann committed:
Update tutorial/path-operation-configuration.md

nilslindemann committed:
Update tutorial/path-params-numeric-validations.md + prompt

There are some instances of using `<code>...</code>` instead of backticks in the English document, which I fixed on the way, as it will not require a retranslation of other languages.

nilslindemann committed:
Update tutorial/path-params.md

nilslindemann committed:
Update tutorial/query-param-models.md

nilslindemann committed:
Update tutorial/query-params-str-validations.md

nilslindemann committed:
Update tutorial/query-params.md + Prompt

The LLM wrote `Pfadparameter`, which is not bad, but let's use or not use a dash unison for all FastAPI specific path operation parameters. As the translation so far used one, I now defined it to be so.

nilslindemann committed:
Intermezzo: Tell LLM to not add anchors to links

As it did so once (like in the example). This is actually super smart, the anchor exists (is the LLM caching the Anchors of headings in other documents?!), but I guess this  possibly creates confusion, so avoid.

nilslindemann committed:
Update tutorial/request-files.md + prompt

Also resort promp slightly and replace a code element with backticks in the English document.

nilslindemann committed:
Update tutorial/request-form-models.md

nilslindemann committed:
Update tutorial/request-forms-and-files.md

nilslindemann committed:
Update tutorial/request-forms.md + prompt

nilslindemann committed:
Intermezzo: Undo changes in two English documents

I guess these html elements were used purposefully because of rendering issues.

nilslindemann committed:
Intermezzo: Use GPT-5

This commit started as an attempt to  handle translation of abbr elements, which I was not able to get to work in a clean way. Shortly before I resigned I had the idea to try `openai:gpt-5` as agent in the translate.py, instead of `openai:gpt-4o` and BAM it worked. A lot of other things also worked much better. Modifications had to be made though because GPT-5 interpreted some rules stricter, so some reformulation and some more examples were needed. Will translate with that agent in the remaining translations to test if these prompts work well.

nilslindemann committed:
Update tutorial/response-model.md

I created this before with GPT-4o, twice as much changes, plus it did not understand some of my rule definitons (Though the prompt was different). So I reverted everything and tried the same with GPT-5. This diff is a joy, only the relevant changes.

nilslindemann committed:
Update tutorial/response-status-code.md

nilslindemann committed:
Update tutorial/schema-extra-example

nilslindemann committed:
Update tutorial/sql-databases.md

nilslindemann committed:
Update tutorial/static-files.md + prompt

nilslindemann committed:
Update tutorial/testing.md

nilslindemann committed:
Update tutorial/dependencies/index.md + prompt

Starting with this commit and in the remaining two folders, I translate the documents in the order they appear in the documentation. Should have done that from the beginning, as it helps the LLM learning things (I guess).

nilslindemann committed:
Update tutorial/dependencies/classes-as-dependencies.md

nilslindemann committed:
Update tutorial/dependencies/sub-dependencies.md + general prompt

nilslindemann committed:
Update dependencies/dependencies-in-path-operation-decorators.md

nilslindemann committed:
Update dependencies/global-dependencies.md

nilslindemann committed:
Update dependencies/dependencies-with-yield.md

It not just kept the translations in the Mermaid diagram (as told) but also updated/fixed them and removed a line of code which was gone in the English diagram.

And fixed/updated many other things. Not one unnecessary change.

nilslindemann committed:
Update tutorial/security/index.md

nilslindemann committed:
Intermezzo: You --> you

Fix a missing space on the way

nilslindemann committed:
Intermezzo - Explain conversion of quotation marks different

because I have the impression that it uses the '``' in the second example and therefore does not see them.

nilslindemann committed:
Update security/first-steps.md

nilslindemann committed:
Intermezzo: `Es testen` --> `Testen`

LLM insists on writing it like this and I am fine with it. It is also written like this in the `/// check | Testen` blocks.

nilslindemann committed:
Update security/get-current-user.md

nilslindemann committed:
Update tutorial/security/simple-oauth2.md

Forgot the `tutorial/` in the messages of the two previous `Update` commits, but it was already pushed. Sorry about that.

nilslindemann committed:
Update tutorial/security/oauth2-jwt.md

Done with the manual walkthrough.

My final temporary git ignore additions (to hide files in VSCode) may be of interest:

# Temporarily ignore unrelated files

/*
!/docs
/docs/*
!/docs/en
!/docs/de
/docs/*/*
!/docs/*/docs
/docs/*/docs/css
/docs/*/docs/img
/docs/*/docs/js
/docs/*/docs/reference
/docs/*/docs/release-notes.md
/docs/*/docs/fastapi-people.md
/docs/*/docs/external-links.md
/docs/*/docs/newsletter.md
/docs/*/docs/management-tasks.md
/docs/*/docs/management.md
/docs/*/docs/contributing.md

# Temporarily ignore done files

/docs/*/docs/alternatives.md
/docs/*/docs/async.md
/docs/*/docs/benchmarks.md
/docs/*/docs/environment-variables.md
/docs/*/docs/fastapi-cli.md
/docs/*/docs/features.md
/docs/*/docs/help-fastapi.md
/docs/*/docs/history-design-future.md
/docs/*/docs/index.md
/docs/*/docs/project-generation.md
/docs/*/docs/python-types.md
/docs/*/docs/virtual-environments.md
/docs/*/docs/about/index.md
/docs/*/docs/about
/docs/*/docs/advanced/additional-responses.md
/docs/*/docs/advanced/additional-status-codes.md
/docs/*/docs/advanced/advanced-dependencies.md
/docs/*/docs/advanced/async-tests.md
/docs/*/docs/advanced/behind-a-proxy.md
/docs/*/docs/advanced/custom-response.md
/docs/*/docs/advanced/dataclasses.md
/docs/*/docs/advanced/events.md
/docs/*/docs/advanced/generate-clients.md
/docs/*/docs/advanced/index.md
/docs/*/docs/advanced/middleware.md
/docs/*/docs/advanced/openapi-callbacks.md
/docs/*/docs/advanced/openapi-webhooks.md
/docs/*/docs/advanced/path-operation-advanced-configuration.md
/docs/*/docs/advanced/response-change-status-code.md
/docs/*/docs/advanced/response-cookies.md
/docs/*/docs/advanced/response-directly.md
/docs/*/docs/advanced/response-headers.md
/docs/*/docs/advanced/settings.md
/docs/*/docs/advanced/sub-applications.md
/docs/*/docs/advanced/templates.md
/docs/*/docs/advanced/testing-dependencies.md
/docs/*/docs/advanced/testing-events.md
/docs/*/docs/advanced/testing-websockets.md
/docs/*/docs/advanced/using-request-directly.md
/docs/*/docs/advanced/websockets.md
/docs/*/docs/advanced/wsgi.md
/docs/*/docs/advanced/security/http-basic-auth.md
/docs/*/docs/advanced/security/index.md
/docs/*/docs/advanced/security/oauth2-scopes.md
/docs/*/docs/advanced/security
/docs/*/docs/advanced
/docs/*/docs/deployment/cloud.md
/docs/*/docs/deployment/concepts.md
/docs/*/docs/deployment/docker.md
/docs/*/docs/deployment/https.md
/docs/*/docs/deployment/index.md
/docs/*/docs/deployment/manually.md
/docs/*/docs/deployment/server-workers.md
/docs/*/docs/deployment/versions.md
/docs/*/docs/deployment
/docs/*/docs/how-to/conditional-openapi.md
/docs/*/docs/how-to/configure-swagger-ui.md
/docs/*/docs/how-to/custom-docs-ui-assets.md
/docs/*/docs/how-to/custom-request-and-route.md
/docs/*/docs/how-to/extending-openapi.md
/docs/*/docs/how-to/general.md
/docs/*/docs/how-to/graphql.md
/docs/*/docs/how-to/index.md
/docs/*/docs/how-to/separate-openapi-schemas.md
/docs/*/docs/how-to/testing-database.md
/docs/*/docs/how-to
/docs/*/docs/learn/index.md
/docs/*/docs/learn
/docs/*/docs/resources/index.md
/docs/*/docs/resources
/docs/*/docs/tutorial/background-tasks.md
/docs/*/docs/tutorial/bigger-applications.md
/docs/*/docs/tutorial/body-fields.md
/docs/*/docs/tutorial/body-multiple-params.md
/docs/*/docs/tutorial/body-nested-models.md
/docs/*/docs/tutorial/body-updates.md
/docs/*/docs/tutorial/body.md
/docs/*/docs/tutorial/cookie-param-models.md
/docs/*/docs/tutorial/cookie-params.md
/docs/*/docs/tutorial/cors.md
/docs/*/docs/tutorial/debugging.md
/docs/*/docs/tutorial/encoder.md
/docs/*/docs/tutorial/extra-data-types.md
/docs/*/docs/tutorial/extra-models.md
/docs/*/docs/tutorial/first-steps.md
/docs/*/docs/tutorial/handling-errors.md
/docs/*/docs/tutorial/header-param-models.md
/docs/*/docs/tutorial/header-params.md
/docs/*/docs/tutorial/index.md
/docs/*/docs/tutorial/metadata.md
/docs/*/docs/tutorial/middleware.md
/docs/*/docs/tutorial/path-operation-configuration.md
/docs/*/docs/tutorial/path-params-numeric-validations.md
/docs/*/docs/tutorial/path-params.md
/docs/*/docs/tutorial/query-param-models.md
/docs/*/docs/tutorial/query-params-str-validations.md
/docs/*/docs/tutorial/query-params.md
/docs/*/docs/tutorial/request-files.md
/docs/*/docs/tutorial/request-form-models.md
/docs/*/docs/tutorial/request-forms-and-files.md
/docs/*/docs/tutorial/request-forms.md
/docs/*/docs/tutorial/response-model.md
/docs/*/docs/tutorial/response-status-code.md
/docs/*/docs/tutorial/schema-extra-example.md
/docs/*/docs/tutorial/sql-databases.md
/docs/*/docs/tutorial/static-files.md
/docs/*/docs/tutorial/testing.md
/docs/*/docs/tutorial/dependencies/index.md
/docs/*/docs/tutorial/dependencies/classes-as-dependencies.md
/docs/*/docs/tutorial/dependencies/sub-dependencies.md
/docs/*/docs/tutorial/dependencies/dependencies-in-path-operation-decorators.md
/docs/*/docs/tutorial/dependencies/global-dependencies.md
/docs/*/docs/tutorial/dependencies/dependencies-with-yield.md
/docs/*/docs/tutorial/dependencies
/docs/*/docs/tutorial/security/index.md
/docs/*/docs/tutorial/security/first-steps.md
/docs/*/docs/tutorial/security/get-current-user.md
/docs/*/docs/tutorial/security/simple-oauth2.md
/docs/*/docs/tutorial/security/oauth2-jwt.md
/docs/*/docs/tutorial/security
/docs/*/docs/tutorial
/docs/*/docs
/docs/*
/docs

nilslindemann committed:
Prepare translate.py for Batch translation

... making a few fixes/changes:

* Fix a deprecation warning
* Fix a probably not intended hardcoded "es" language identifier when calling `translate_page`
* Fix `iter_en_paths_to_translate` and `non_translated_sections ` to also work on Windows, which uses backslashes in stringified paths
* Add `mode`, `verbose` and `preview` parameters to `translate_lang`, add a helper function to print path infos, and improve its output to show the progress.
* Add documentation to some function parameters

nilslindemann committed:
Intermezzo `z.B.` --> `z. B.`

Forgot to include the change to the German prompt. It is in the later commit.

nilslindemann committed:
German docs batch translation

A complete run through the German docs to test how the prompts work "in production".

gpt-5 found a lot of things which gpt-4o and I have overlooked.

I had to manually modify/fix approximately 10 changes it suggested, which is a good result for 111 files.

The German docs are now completely translated and in sync.

nilslindemann committed:
Two Fixes in English documents

Fond these while manually checking the previous batch translation commit

nilslindemann committed:
Changes to German prompt reflecting results from batch translation

* (Hopefully) improve rule when not to convert to typographic quotation marks, re-add some examples
* Refine rule how to translate abbr elements
* Include and remove some words

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

nilslindemann committed:
Fix two things in the German prompt

nilslindemann committed:
Merge branch 'update-prompts-and-german-translation' of https://github.com/nilslindemann/fastapi into update-prompts-and-german-translation

nilslindemann committed:
A few more changes

... made while re-reading the diff until inclusive docs/de/docs/deployment/cloud.md

One change is an addition of a sentence in an English dokument.

Two word additions to the German prompt

nilslindemann committed:
Fix headings which have periods

And add a rule to the prompt.

nilslindemann committed:
Use long dashes in headings

nilslindemann committed:
Add space between ellipsis and words

Also added rule to prompt.

I thought it is okay, meanwhile, in German when there is no space inbetween, (which is why i did not fix these changes before) but this is actually not the case. In English both spellings are okay.

Has one change in the English document. If we leave those backticks, the LLM should by definition not translate it, but we want this here.

nilslindemann committed:
Intermezzo: `Testen` --> `Es testen`

nilslindemann committed:
Intermezzo: `als`, `über` --> `via`

nilslindemann committed:
Intermezzo: `Exception-Handler` --> `Exceptionhandler`

nilslindemann committed:
Intermezzo: `deprecated` --> `deprecatet`

That is how such english words are spelled in German, they end with `t` (which I didnt know back then when I translated).

And wrap it in an abbr which explains the meaning.

nilslindemann committed:
Misc fixes

This and the previous six commits are changes which I did while going through the diff of the PR until inclusive tutorial/handling-errors.md

nilslindemann committed:
Update prompts

nilslindemann committed:
Add LLM test file

nilslindemann committed:
Update index.md

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

nilslindemann committed:
Intermezzo: `Anfrage` --> `Request`

First `Request` per page gets ann abbr with an explanation.

nilslindemann committed:
Intermezzo: `Antwort` --> `Response`

As previous commit

nilslindemann committed:
Intermezzo: `Abfrage` --> `Query`

nilslindemann committed:
Merge branch 'update-prompts-and-german-translation' of https://github.com/nilslindemann/fastapi into update-prompts-and-german-translation

nilslindemann committed:
Intermezzo: `dateiähnlich` --> `dateiartig`

And linkify `file-like`, also in English document. It is the same link used in the other English document which uses that term.

nilslindemann committed:
Intermezzo: enrich `Dictionary`

Wrap first occurrence into explaining abbr. Translate closer to the english text (`dict` vs `dictionary`)

nilslindemann committed:
Intermezzo: Enum

nilslindemann committed:
Misc fixes and changes

Changes until tutorial/query-params.md

nilslindemann committed:
Update Twitter

nilslindemann committed:
Update async.md

nilslindemann committed:
Squashed commit of the following:

commit 176cd8c9ef81c3e4c5ba826e7eb87554bae17080
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 15:20:12 2025 +0000

    📝 Update release notes

    [skip ci]

commit 1884d76f613d951e2243e50a3df4d9a479f00a9d
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 17:19:49 2025 +0200

    🔧 Update sponsors: remove Platform.sh (#14027)

commit 8062aabdaa4e39a31bd01f3fb35428a3207de1bb
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 15:02:29 2025 +0000

    📝 Update release notes

    [skip ci]

commit ee9ccac1e5f5faca45a56e0e0eea109dfb2988a4
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 17:02:08 2025 +0200

    🔧 Update sponsors: remove Mobb (#14026)

commit 5cd4c3b6bd3eec27fa3c621e4a2d2a2c00605057
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:59:17 2025 +0000

    📝 Update release notes

    [skip ci]

commit 4584f706bdf833ede7e6d5771c0d16bb9c01fafd
Author: Jom Karlo Verzosa <jomkarlov@gmail.com>
Date:   Sun Aug 31 18:58:56 2025 +0800

    📝 Add deprecation info block about `dict()` in `docs/tutorial/body.md` (#13906)

    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit ba9c8fba0b26beccfc1300f9da1f4dea97e08157
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:50:12 2025 +0000

    📝 Update release notes

    [skip ci]

commit d9249c1949b856f60bd2a79c94e43c61989a0af9
Author: Valentyn <valentin.druzhinin@gmail.com>
Date:   Sun Aug 31 06:49:48 2025 -0400

    📝 Fix Twitter to be X (Twitter) everywhere in documentation (#13809)

    Co-authored-by: Valentyn Druzhynin <v.druzhynin@zakaz.global>
    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit a973e787af25d4dc87ddd15ad2c0819437d08f6f
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:33:32 2025 +0000

    📝 Update release notes

    [skip ci]

commit 1088d2abd908b7101ad7ea4ed0a5e141307d7693
Author: Ashish Pandey <126683810+Ashish-Pandey62@users.noreply.github.com>
Date:   Sun Aug 31 16:17:57 2025 +0545

    🐛 Prevent scroll-to-top on restart/fast buttons in `termynal.js` (#13714)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit 98ec6a6079f31aaf95814e1c1d6194d12bd9aaa9
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:29:48 2025 +0000

    📝 Update release notes

    [skip ci]

commit 6b4d292f3a95709821429ee1b3dead1563c91c75
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:29:27 2025 +0000

    📝 Update release notes

    [skip ci]

commit d4ddcc5878a6cccd9815e76e69ff1fb52bd52457
Author: z0z0r4 <z0z0r4@outlook.com>
Date:   Sun Aug 31 18:29:21 2025 +0800

    📝 Update testing events documentation (#13259)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
    Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>
    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit 0e5832aa6d8c9badd203c2d8be7ae1842bb3c9e4
Author: Hotah Ma <mahongtao0x7e0@163.com>
Date:   Sun Aug 31 18:29:01 2025 +0800

    📝 Remove obsolete `url` field in error responses in docs (#13655)

    Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

commit ee2acd8abc7b63863be3ee6ef61e46878d3549b9
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:03:35 2025 +0000

    📝 Update release notes

    [skip ci]

commit e902ed5fc68dc73b4fbc5ce3c4434e5c493683ee
Author: Arnaud Durand <arnaud.durand@live.com>
Date:   Sun Aug 31 12:03:10 2025 +0200

    📝 Bring the `scope` claim in line with the standard in `docs_src/security/tutorial005.py` (#11189)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
    Co-authored-by: Yurii Motov <yurii.motov.monte@gmail.com>

commit cef1f166dfe326792001e2cfe0dfeeebe45e7cd3
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:59:28 2025 +0000

    📝 Update release notes

    [skip ci]

commit 8e63f75919f1aca30d5dc62cfcf0d8c027e8c8cd
Author: Soul Lee <alus20x@gmail.com>
Date:   Sun Aug 31 18:59:07 2025 +0900

    📝 Update TrustedHostMiddleware Documentation (#11441)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
    Co-authored-by: Alejandra <90076947+alejsdev@users.noreply.github.com>
    Co-authored-by: Sofie Van Landeghem <svlandeg@users.noreply.github.com>
    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>
    Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

commit e1b9cc00caa9729880cdea838706a22797932d31
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:56:48 2025 +0000

    📝 Update release notes

    [skip ci]

commit 408b8a9bebf3af84943acf7b57bf91b0c5e76a25
Author: Denny Biasiolli <dennybiasiolli@users.noreply.github.com>
Date:   Sun Aug 31 11:56:21 2025 +0200

    📝 Remove links to site callbackhell.com that doesn't exist anymore (#14006)

commit 0817c955ec88cac09a8aec87b27c5acd14e65861
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:16:03 2025 +0000

    📝 Update release notes

    [skip ci]

commit c55f7138a11e9d315db95c1a29711b0d36baa33d
Author: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>
Date:   Sun Aug 31 11:15:41 2025 +0200

    📝 Add permalinks to headers in English docs (#13993)

    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit 7653de2715c17feddfdc5621f9097aaae3e374b1
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:11:36 2025 +0000

    📝 Update release notes

    [skip ci]

commit 784f068abaa57aa25eac104e6d831ca12980f90e
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 11:11:15 2025 +0200

    🛠️ Update `mkdocs_hooks` to handle headers with permalinks when building docs (#14025)

    Co-authored-by: Yurii Motov <yurii.motov.monte@gmail.com>

commit 6db05770f6f42d6c3e96e5e228b8fc7447026ce3
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Mon Aug 25 20:03:24 2025 +0000

    📝 Update release notes

    [skip ci]

commit 6be02e3d5261b1341a76d0e0f337ec5aa78f4e62
Author: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Date:   Mon Aug 25 22:03:02 2025 +0200

    ⬆ [pre-commit.ci] pre-commit autoupdate (#14016)

    updates:
    - [github.com/astral-sh/ruff-pre-commit: v0.12.9 → v0.12.10](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.9...v0.12.10)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit 9cf7b70d7b282b0671dbb1a93a4cbc31aac58abe
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Wed Aug 20 09:11:20 2025 +0000

    📝 Update release notes

    [skip ci]

commit f75c1532f612137795b291f42fb507f7f103807b
Author: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>
Date:   Wed Aug 20 11:10:51 2025 +0200

    ⬆ Bump `mkdocs-macros-plugin` from 1.3.7 to 1.3.9 (#14003)

    Bump mkdocs-macros-plugin from 1.3.7 to 1.3.9

commit 5c3a70d5b6def6f5ee471c3f9c7e6d2b2c6f6200
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Mon Aug 18 21:07:25 2025 +0000

    📝 Update release notes

    [skip ci]

commit 6a45249303cf5ce8c9cde2c35497d2d12a4221a1
Author: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Date:   Mon Aug 18 23:07:04 2025 +0200

    ⬆ [pre-commit.ci] pre-commit autoupdate (#13999)

    updates:
    - [github.com/astral-sh/ruff-pre-commit: v0.12.8 → v0.12.9](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.8...v0.12.9)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit c23051682bbeb254faee28fe80beeeab1764f7c4
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Mon Aug 18 06:35:06 2025 +0000

    📝 Update release notes

    [skip ci]

commit df779885fa094da506193a3772c6ab0335728790
Author: Mika <44454249+anfreshman@users.noreply.github.com>
Date:   Mon Aug 18 14:34:40 2025 +0800

    📝 Fix code include for Pydantic models example in `docs/zh/docs/python-types.md` (#13997)

    Updated the Pydantic expiration example in the Chinese documentation

nilslindemann committed:
Update prompts

nilslindemann committed:
Misc changes

nilslindemann committed:
Squashed commit of the following:

commit 3e2dbf91699ef2e3a713372f099ef278b3bcc59e
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 19:34:37 2025 +0000

    📝 Update release notes

    [skip ci]

commit f5b77ff0fcd8e191d9be98d61292ecfaa6cb89a6
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 21:34:08 2025 +0200

    📝 Add documentation for Behind a Proxy - Proxy Forwarded Headers, using `--forwarded-allow-ips="*"` (#14028)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit 176cd8c9ef81c3e4c5ba826e7eb87554bae17080
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 15:20:12 2025 +0000

    📝 Update release notes

    [skip ci]

commit 1884d76f613d951e2243e50a3df4d9a479f00a9d
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 17:19:49 2025 +0200

    🔧 Update sponsors: remove Platform.sh (#14027)

commit 8062aabdaa4e39a31bd01f3fb35428a3207de1bb
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 15:02:29 2025 +0000

    📝 Update release notes

    [skip ci]

commit ee9ccac1e5f5faca45a56e0e0eea109dfb2988a4
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 17:02:08 2025 +0200

    🔧 Update sponsors: remove Mobb (#14026)

commit 5cd4c3b6bd3eec27fa3c621e4a2d2a2c00605057
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:59:17 2025 +0000

    📝 Update release notes

    [skip ci]

commit 4584f706bdf833ede7e6d5771c0d16bb9c01fafd
Author: Jom Karlo Verzosa <jomkarlov@gmail.com>
Date:   Sun Aug 31 18:58:56 2025 +0800

    📝 Add deprecation info block about `dict()` in `docs/tutorial/body.md` (#13906)

    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit ba9c8fba0b26beccfc1300f9da1f4dea97e08157
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:50:12 2025 +0000

    📝 Update release notes

    [skip ci]

commit d9249c1949b856f60bd2a79c94e43c61989a0af9
Author: Valentyn <valentin.druzhinin@gmail.com>
Date:   Sun Aug 31 06:49:48 2025 -0400

    📝 Fix Twitter to be X (Twitter) everywhere in documentation (#13809)

    Co-authored-by: Valentyn Druzhynin <v.druzhynin@zakaz.global>
    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit a973e787af25d4dc87ddd15ad2c0819437d08f6f
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:33:32 2025 +0000

    📝 Update release notes

    [skip ci]

commit 1088d2abd908b7101ad7ea4ed0a5e141307d7693
Author: Ashish Pandey <126683810+Ashish-Pandey62@users.noreply.github.com>
Date:   Sun Aug 31 16:17:57 2025 +0545

    🐛 Prevent scroll-to-top on restart/fast buttons in `termynal.js` (#13714)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit 98ec6a6079f31aaf95814e1c1d6194d12bd9aaa9
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:29:48 2025 +0000

    📝 Update release notes

    [skip ci]

commit 6b4d292f3a95709821429ee1b3dead1563c91c75
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:29:27 2025 +0000

    📝 Update release notes

    [skip ci]

commit d4ddcc5878a6cccd9815e76e69ff1fb52bd52457
Author: z0z0r4 <z0z0r4@outlook.com>
Date:   Sun Aug 31 18:29:21 2025 +0800

    📝 Update testing events documentation (#13259)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
    Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>
    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit 0e5832aa6d8c9badd203c2d8be7ae1842bb3c9e4
Author: Hotah Ma <mahongtao0x7e0@163.com>
Date:   Sun Aug 31 18:29:01 2025 +0800

    📝 Remove obsolete `url` field in error responses in docs (#13655)

    Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

commit ee2acd8abc7b63863be3ee6ef61e46878d3549b9
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 10:03:35 2025 +0000

    📝 Update release notes

    [skip ci]

commit e902ed5fc68dc73b4fbc5ce3c4434e5c493683ee
Author: Arnaud Durand <arnaud.durand@live.com>
Date:   Sun Aug 31 12:03:10 2025 +0200

    📝 Bring the `scope` claim in line with the standard in `docs_src/security/tutorial005.py` (#11189)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
    Co-authored-by: Yurii Motov <yurii.motov.monte@gmail.com>

commit cef1f166dfe326792001e2cfe0dfeeebe45e7cd3
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:59:28 2025 +0000

    📝 Update release notes

    [skip ci]

commit 8e63f75919f1aca30d5dc62cfcf0d8c027e8c8cd
Author: Soul Lee <alus20x@gmail.com>
Date:   Sun Aug 31 18:59:07 2025 +0900

    📝 Update TrustedHostMiddleware Documentation (#11441)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
    Co-authored-by: Alejandra <90076947+alejsdev@users.noreply.github.com>
    Co-authored-by: Sofie Van Landeghem <svlandeg@users.noreply.github.com>
    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>
    Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

commit e1b9cc00caa9729880cdea838706a22797932d31
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:56:48 2025 +0000

    📝 Update release notes

    [skip ci]

commit 408b8a9bebf3af84943acf7b57bf91b0c5e76a25
Author: Denny Biasiolli <dennybiasiolli@users.noreply.github.com>
Date:   Sun Aug 31 11:56:21 2025 +0200

    📝 Remove links to site callbackhell.com that doesn't exist anymore (#14006)

commit 0817c955ec88cac09a8aec87b27c5acd14e65861
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:16:03 2025 +0000

    📝 Update release notes

    [skip ci]

commit c55f7138a11e9d315db95c1a29711b0d36baa33d
Author: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>
Date:   Sun Aug 31 11:15:41 2025 +0200

    📝 Add permalinks to headers in English docs (#13993)

    Co-authored-by: Sebastián Ramírez <tiangolo@gmail.com>

commit 7653de2715c17feddfdc5621f9097aaae3e374b1
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Sun Aug 31 09:11:36 2025 +0000

    📝 Update release notes

    [skip ci]

commit 784f068abaa57aa25eac104e6d831ca12980f90e
Author: Sebastián Ramírez <tiangolo@gmail.com>
Date:   Sun Aug 31 11:11:15 2025 +0200

    🛠️ Update `mkdocs_hooks` to handle headers with permalinks when building docs (#14025)

    Co-authored-by: Yurii Motov <yurii.motov.monte@gmail.com>

commit 6db05770f6f42d6c3e96e5e228b8fc7447026ce3
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Mon Aug 25 20:03:24 2025 +0000

    📝 Update release notes

    [skip ci]

commit 6be02e3d5261b1341a76d0e0f337ec5aa78f4e62
Author: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Date:   Mon Aug 25 22:03:02 2025 +0200

    ⬆ [pre-commit.ci] pre-commit autoupdate (#14016)

    updates:
    - [github.com/astral-sh/ruff-pre-commit: v0.12.9 → v0.12.10](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.9...v0.12.10)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit 9cf7b70d7b282b0671dbb1a93a4cbc31aac58abe
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Wed Aug 20 09:11:20 2025 +0000

    📝 Update release notes

    [skip ci]

commit f75c1532f612137795b291f42fb507f7f103807b
Author: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>
Date:   Wed Aug 20 11:10:51 2025 +0200

    ⬆ Bump `mkdocs-macros-plugin` from 1.3.7 to 1.3.9 (#14003)

    Bump mkdocs-macros-plugin from 1.3.7 to 1.3.9

commit 5c3a70d5b6def6f5ee471c3f9c7e6d2b2c6f6200
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Mon Aug 18 21:07:25 2025 +0000

    📝 Update release notes

    [skip ci]

commit 6a45249303cf5ce8c9cde2c35497d2d12a4221a1
Author: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Date:   Mon Aug 18 23:07:04 2025 +0200

    ⬆ [pre-commit.ci] pre-commit autoupdate (#13999)

    updates:
    - [github.com/astral-sh/ruff-pre-commit: v0.12.8 → v0.12.9](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.8...v0.12.9)

    Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

commit c23051682bbeb254faee28fe80beeeab1764f7c4
Author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Date:   Mon Aug 18 06:35:06 2025 +0000

    📝 Update release notes

    [skip ci]

commit df779885fa094da506193a3772c6ab0335728790
Author: Mika <44454249+anfreshman@users.noreply.github.com>
Date:   Mon Aug 18 14:34:40 2025 +0800

    📝 Fix code include for Pydantic models example in `docs/zh/docs/python-types.md` (#13997)

    Updated the Pydantic expiration example in the Chinese documentation

nilslindemann committed:
Translate recent additions

nilslindemann committed:
Fix accidental duplication of heading

nilslindemann committed:
Merge branch 'master' into update-prompts-and-german-translation

nilslindemann committed:
Put typographic quotes inside the abbr

nilslindemann committed:
Merge branch 'update-prompts-and-german-translation' of https://github.com/nilslindemann/fastapi into update-prompts-and-german-translation

nilslindemann committed:
Update help-fastapi.md

nilslindemann committed:
Update features.md

nilslindemann committed:
Update translate.py

nilslindemann committed:
Temp

nilslindemann committed:
` - ` --> ` – `

nilslindemann committed:
Update llm-prompt.md

nilslindemann committed:
Update index.md

nilslindemann committed:
`FastAPI-CLI` --> `FastAPI CLI`

nilslindemann committed:
`zu` --> `auf`

nilslindemann committed:
Typo

nilslindemann reviewed (commented):
Reviewed it, making some changes on the way. @YuriiMotov can you take a look at the changes in the `translate.py` and test? Does it work with other languages? Is it too verbose? If okay, can you apply it to #13968? Same for @tiangolo regarding the German prompt. Is it okay? Too verbose? If okay, can you apply it to ##13931? @alejsdev I see you have added eyes to my first post, which I interpret as, you are the one planning to review it? My hero :-)

However (also @tiangolo), here is the deal: The PR is huge, and I can understand when no one likes to review it. I have made it, and then I have reviewed it, actually twice, as I also reviewed the translations back them, when I created them. I can guarantee that there are very few errors in the translation and that it is in sync. If there are issues in style, then that can be handled in a new PR. So my suggestion is to focus on the changes in the prompts and in the English docs, and to just randomly test a few German translations, and otherwise trust me, The LLM will anyway do what it wants whenever there is a retranslation, and we have to obey its suggestions a bit (or make the changes to the English documents), to get a clean diff. Therefore, too much style nitpicking is not productive, it is better to improve the English document or the prompts.

I am making a break today, let's see if I have the energy tomorrow to write a script which compares things like links, amount of paragraphs, code blocks, color boxes, tab blocks, hashes in headings, ...

YuriiMotov commented on scripts/translate.py:
```diff
@@ -182,39 +783,72 @@ def iter_all_en_paths() -> Iterable[Path]:
+    mode: Annotated[
+        str,
+        typer.Option(
+            help="Which files of the target language to translate, one of: `missing`, `existing`, `all`"
+        ),
+    ] = "missing",
```
I think mode should be: `missing`, `updated`, `all`.
And it should be `all` by default

nilslindemann replied:
Lets outsource this one too.

YuriiMotov reviewed (commented):
@nilslindemann, thank you for working on this!

I ran translation script to translate several pages to Russian. It works good overall!

Sometimes it fails to follow rules, but I don't think we need to spend more time on polishing the prompt - it's just stubborn LLMs.. They will always mess up from time to time..

Prompt is huge, but at the same time it's now better structured and should work better.
I think we should trust @nilslindemann (I believe he made a lot of experiments before coming up with this version) and go with this version of the prompt.

Let's discuss a few things I commented and then I will move things related to `translate.py` script to https://github.com/fastapi/fastapi/pull/13968

nilslindemann committed:
Update docs/en/docs/tutorial/debugging.md

Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

nilslindemann commented on scripts/translate.py:
```diff
@@ -96,30 +685,42 @@ def generate_en_path(*, lang: str, path: Path) -> Path:
+            envvar="EN_PATH",
+            help="Path to the English source, relative to the FastAPI root directory. If not given, `docs/en/docs/_llm-test.md` is used.",
+        ),
+    ] = Path("docs/en/docs/_llm-test.md"),
```
Yeah, this may result in running a translation unintended. Will remove.

nilslindemann replied:
(waiting for tiangolo's verdict regarding removal of this function)

nilslindemann committed:
Put definitions in quotes

... like in the English prompt.

nilslindemann committed:
Typo and singular

nilslindemann committed:
Remove special quotes around schemes

They may get interpreted "too" literally.

nilslindemann committed:
Merge branch 'update-prompts-and-german-translation' of https://github.com/nilslindemann/fastapi into update-prompts-and-german-translation

nilslindemann committed:
Remove added default

It may lead to accidental translation. It better should raise when this parameter is missing.

nilslindemann committed:
Update German prompt and prompt test file

nilslindemann committed:
Update/fix German prompt and test files

Cleanup for the abbr issue, otherwise unrelated.

nilslindemann committed:
Improve instructions for existing translations

I noticed that it forgot to remove a removed sentence and to add an added part of a sentence, when I reordered/updated a few sentences in the test file. It was stubborn even when retranslating. After these reformulations and the the added sentence at the end it saw these changes and properly synced them.

This is also cleanup for the abbr issue, not directly related.

nilslindemann committed:
Clarify an instruction

Also cleanup for the abbr issue, not directly related.

nilslindemann committed:
Add a definition

I was not sure if it understands the difference between an HTML element and a HTML tag.

Also not directly related to the abbr issue.

nilslindemann committed:
Improve abbr handling

After these changes it now seems to handle descriptions in title attributes well, both when translating a document the first time – see the abbr section in the `ru/docs/_llm-test.md` which I committed for reference, but I am not sure if everything is correct there, as I do not speak Russian – and when updating translations – tested with e.g. `ru/docs/features.md` and `ru/docs/tutorial/path-params-numeric-validations.md`, not committed.

I simplified rule 3 to just match when the separator is a colon (`:`), not when it is a comma (`,`), there are otherwise too many cases it may misunderstand, as commas are regularly used in title attributes. Updated the two instances in the English docs (`features.md` and `tutorial\sql-databases.md`) where a comma was used instead of a colon in a case, which we match with rule 3. I fixed the results in schemes they accidentally had `(de)` appended, but they are language independent.

nilslindemann committed:
`ref` --> `Ref.`

Found while testing if it keeps added abbr's after these recent commits (it does).

It wants to write these small, but they are written big in German and end with a dot (as I just found out).

nilslindemann commented on scripts/translate.py:
```diff
@@ -22,55 +24,642 @@
+
+Translate HTML abbr elements as follows:
+
+1) If the title attribute gives the full phrase for an abbreviation, then keep the phrase, append a dash («–»), followed by the translation of the phrase.
```
Works now.

nilslindemann commented:
@YuriiMotov the abbr issue seems to be fixed now, let me know if you find other things.

YuriiMotov commented on scripts/translate.py:
```diff
@@ -128,16 +743,19 @@ def translate_page(
+                "There is an existing previous translation for the original English content, that may be outdated.",
                 "Update the translation only where necessary:",
-                "- If the original English content has changed, reflect that in the translation.",
+                "- If the original English content has added parts, also add these parts to the translation.",
+                "- If the original English content has removed parts, also remove them from the translation, unless you were instructed earlier to not do that in specific cases.",
+                "- If parts of the original English content have changed, also change those parts in the translation.",
```
I think this is a tricky part.
We don't provide old version of En document.
So, LLM can't distinguish between two cases:
* part of the sentence was removed from the original document
* sentence in the original document remains unchanged, but translation contains additional parts added by human editor (and we instruct LLM to preserve such human edits)

I'm afraid these instructions will not be followed. So, maybe leave the shorter version?

YuriiMotov replied:
We can probably use some kind of markers to mark such human editor's additions.

Original:
```
This is original text of the sentence
```

Translation:
```
This is original text <!-- PRESERVE --> (modified a bit) <!-- END PRESERVE --> of the sentence
```
Or:
```
This is original text <span class="human-edit"> (modified a bit) </span> of the sentence
```

YuriiMotov replied:
When you refer to the last sentence, do you mean this?
> UNLESS you were instructed earlier to behave different, there MUST NOT be whole sentences or partial sentences in the updated translation, which are not in the original English content, and there MUST NOT be whole sentences or partial sentences in the original English content, which are not in the updated translation. Remember: the updated translation shall be IN SYNC with the original English content.

nilslindemann replied:
Yes, I meant this sentence. It takes a while to load for such a link when the diff is big in the GitHub GUI.

nilslindemann commented on scripts/translate.py:
```diff
@@ -182,39 +783,72 @@ def iter_all_en_paths() -> Iterable[Path]:
+        ),
+    ] = "missing",
+    verbose: Annotated[bool, typer.Option(help="Print all paths")] = False,
+    preview: Annotated[
```
@YuriiMotov, is it okay for you if I do this and the below issue in a separate PR after this one and the PRs above are resolved? After all, this change does not remove any functionality, so nothing breaks for the moment.

nilslindemann committed:
Merge branch 'fastapi:master' into update-prompts-and-german-translation

nilslindemann committed:
Do this in another PR

Co-Authored-By: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

nilslindemann commented on docs/en/docs/tutorial/debugging.md:
````diff
@@ -62,7 +62,7 @@ from myapp import app
 ```
 
-in that case, the automatically created variable inside of `myapp.py` will not have the variable `__name__` with a value of `"__main__"`.
+in that case inside of `myapp.py` the automatically created variable `__name__` will not have the value `"__main__"`.
````
Reads better, have applied it.

nilslindemann committed:
Apply formulation by Yurii

Forgot/overlooked this one.

Co-Authored-By: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

nilslindemann committed:
Three small changes in translate.py

Before running a retranslation.

nilslindemann committed:
Another full retranslation

... of all 111 documents. These are the unmodified changes made by the LLM. I address some issues in the following commits

nilslindemann committed:
Modifications made by me

nilslindemann committed:
Modifications in the general prompt

Exchange rule 4 and 5 and then exclude additional abbr's (abbr's without an english equivalent) from the list of abbr's which shall always be updated, as the LLM has nothing to diff against in this case. There were cases where it removed good information, which I had to undo, see e.g. the changes in `tutorial\query-param-models.md` in the two previous commits.

nilslindemann committed:
Lax some preferred words in the German prompt

* "Die API-App abliefern" sounds funny, allow "Die API-Anwendung abliefern"
* "Leistung" instead of "Performanz" is usually smoother, we just dont want "Performance" (which is an artistic act) instead of "Performanz".

nilslindemann committed:
Improve handling of string literals in code snippets

It should replace «"car"» with «„Auto“» but leave «`"car"`» as is (and not convert it to e.g. «„car“»). After this change it seems to finally have grasped that rule, at least according to my tests in a few documents, where it previously seemingly randomly did this wrong.

nilslindemann committed:
Undo changes in English docs

These changes are now handled in fixes-in-english-docs (#14048).

nilslindemann committed:
Undo unrelated changes in translate.py

Let's handle these changes in another PR.

Except the modifications in the prompt, this translate.py is now in sync with the translate.py in update-translations-prompt (#13968).

nilslindemann committed:
Merge branch 'fastapi:master' into update-prompts-and-german-translation

nilslindemann committed:
Remove Russian Language prompt

I needed it to run test Russian translations.

nilslindemann committed:
Merge branch 'fastapi:master' into update-prompts-and-german-translation

nilslindemann commented:
Thanks Yurii, good catch.

nilslindemann committed:
Add a period and fix an unclear formulation

nilslindemann committed:
Merge branch 'fastapi:master' into update-prompts-and-german-translation

tiangolo committed:
Merge branch 'master' into update-prompts-and-german-translation

tiangolo reviewed (approved):
Amazing, fantastic job @nilslindemann, thanks a lot! 🙌 

And also thanks @YuriiMotov for all the help with this. 🍰

nilslindemann commented:
Awesome, always nice to have such a big PR done, I hope it was not too much work to review, Tiangolo. Also thanks Yurii.
